samcli/commands/local/lib/swagger/parser.py

```diff
@@ -137,14 +137,20 @@ def get_authorizers(self, event_type: str = Route.API) -> Dict[str, Authorizer]:
                 )
 
             enable_simple_response = authorizer_object.get(SwaggerParser._AUTHORIZER_SIMPLE_RESPONSES, False)
-            if event_type != Route.HTTP or payload_version != LambdaAuthorizer.PAYLOAD_V2 and enable_simple_response:
+
+            if (
+                event_type != Route.HTTP
+                or payload_version != LambdaAuthorizer.PAYLOAD_V2
+                or not isinstance(enable_simple_response, bool)
+            ):
                 enable_simple_response = False
 
-                LOG.warning(
-                    "Simple responses are only available on HTTP APIs with payload version "
-                    "2.0, ignoring for Lambda authorizer '%s'",
-                    auth_name,
-                )
+                if authorizer_object.get(SwaggerParser._AUTHORIZER_SIMPLE_RESPONSES) is not None:
```

Comment:
minor nit: `if authorizer_object.get(SwaggerParser._AUTHORIZER_SIMPLE_RESPONSES, None)` might be enough and more pythonic. Not blocking.

Reply:
Yup, the default return value is `None`, so I decided to leave that unpopulated. As for the `is not None` check, this is here since the customer can provide `0` or `""` and the statement would be false, so `None` is explicitly checked here.